### PR TITLE
hotfix for PT1.6 and torchtext

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

```
Installing collected packages: torch, torchtext
  WARNING: The scripts convert-caffe2-to-onnx and convert-onnx-to-caffe2 are installed in '/home/AzDevOps_azpcontainer/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchvision 0.7.0 requires torch==1.6.0, but you have torch 1.7.1 which is incompatible.
```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Comment by Borda** [comment]
the remaining failers are unrelated to this change
also, this dependency shall be properly resolved in the docker build

**Comment by Borda** [comment]
so the last missing is TPU @tchaton

**Review by s-rog — APPROVED** [review]

File: dockers/tpu-tests/Dockerfile [hunk 1]
```diff
@@ -29,6 +29,9 @@ RUN cd pytorch-lightning && \
     unzip -o legacy/checkpoints.zip -d legacy/ && \
     ls -l legacy/checkpoints/
 
+# todo: torchtext
```
Comment (akihironitta) [inline dockers/tpu-tests/Dockerfile:32]: maybe?
```suggestion
# todo: unpin torchtext version
```

Comment (Borda) [reply dockers/tpu-tests/Dockerfile:32]: turned to be unrelated and skip this edit

File: dockers/tpu-tests/Dockerfile [hunk 2]
```diff
@@ -29,6 +29,9 @@ RUN cd pytorch-lightning && \
...
+RUN pip install torchtext==0.6.0 -U
+
```
Comment (Borda) [inline dockers/tpu-tests/Dockerfile:34]:
```suggestion
```

**Comment by Borda** [comment]
```
 # TorchVision hotfix https://github.com/pytorch/vision/issues/1938
 from syft.util import get_root_data_path
 from six.moves import urllib
 opener = urllib.request.build_opener()
 opener.addheaders = [(\"User-agent\", \"Mozilla/5.0\")]
 urllib.request.install_opener(opener)
```

File: setup.cfg [hunk 3]
```diff
@@ -67,6 +67,8 @@ exclude =
     *.egg
     build
     temp
+    # todo: some strange failure
+    ./tests/callbacks/test_pruning.py
```
Comment (carmocca) [inline setup.cfg:71]: Any idea about this? I can have a look... but should have anything to do with torchtext 🤔

Comment (Borda) [reply setup.cfg:71]: no, it is unrelated and even the check details can't be found in codebase

**Comment by Borda** [comment]
@tchaton the download shall be simply fixed by catching download fails
https://github.com/PyTorchLightning/pytorch-lightning/blob/6166f46281d77830cf88b1cb8dc050c0312fcfcf/pl_examples/basic_examples/autoencoder.py#L29

**Pull request merged**